lib/datadog/core/configuration/option.rb

```diff
@@ -8,17 +8,33 @@ module Configuration
       class Option
         attr_reader :definition
 
-        # Option setting precedence. Higher number means higher precedence.
+        # Option setting precedence.
         module Precedence
+          # Represents an Option precedence level.
+          # Each precedence has a `numeric` value; higher values means higher precedence.
+          # `name` is for inspection purposes only.
+          Value = Struct.new(:numeric, :name) do
+            include Comparable
+
+            def <=>(other)
+              return nil unless other.is_a?(Value)
+
+              numeric <=> other.numeric
+            end
+          end
+
           # Remote configuration provided through the Datadog app.
-          REMOTE_CONFIGURATION = [2, :remote_configuration].freeze
+          REMOTE_CONFIGURATION = Value.new(2, :remote_configuration).freeze
 
           # Configuration provided in Ruby code, in this same process.
-          PROGRAMMATIC = [1, :programmatic].freeze
+          PROGRAMMATIC = Value.new(1, :programmatic).freeze
 
           # Configuration that comes either from environment variables,
           # or fallback values.
-          DEFAULT = [0, :default].freeze
+          DEFAULT = Value.new(0, :default).freeze
+
+          # All precedences, sorted from highest to lowest
+          LIST = [REMOTE_CONFIGURATION, PROGRAMMATIC, DEFAULT].freeze
```

Comment:
Minor: Maybe call `sort` here to enforce that the list follows the defined order? 

```suggestion
          LIST = [REMOTE_CONFIGURATION, PROGRAMMATIC, DEFAULT].sort.reverse.freeze
```